packages/authentication/ios/Plugin/Classes/Results/CheckAppTrackingTransparencyPermissionResult.swift

```diff
@@ -0,0 +1,31 @@
+import Capacitor
+import AppTrackingTransparency
+
+@objc public class CheckAppTrackingTransparencyPermissionResult: NSObject, Result {
+    let status: ATTrackingManager.AuthorizationStatus
+
+    init(_ status: ATTrackingManager.AuthorizationStatus) {
+        self.status = status
+    }
+
+    func toJSObject() -> AnyObject {
+        var result = JSObject()
+        result["status"] = convertStatusToPermissionStatus()
+        return result as AnyObject
+    }
+
+    private func convertStatusToPermissionStatus() -> String {
```

Comment:
```suggestion
        result["status"] = convertStatusToPermissionStatus(status)
        return result as AnyObject
    }

    private func convertStatusToPermissionStatus(_ status: ATTrackingManager.AuthorizationStatus) -> String {
```

Looks less like magic. ^^